### DISCUSSION TITLE
rehypeCitation options.noCite, rehype plugins options

### DISCUSSION THREAD

**Discussion by alexsalce** [timlrx/tailwind-nextjs-starter-blog#997] [Q&A] [answered]
Hi,

I am trying to display citations from my .bib file that aren't cited anywhere in the mdx blog body. 
rehypeCitation documentation indicates this can be accomplished using the `options.noCite` option.
I'm not sure how to use the option, is there any straightforward way to utilize this option?
And maybe more generally, is there a way to use rehype plugins options?  I may have missed some documentation.
Also, I'm still pretty new to nextjs/tailwind so I apologize if I am missing some common knowledge.

Thank you
Alex

**Comment by alexsalce** [comment]
options.noCite
Type: string[].

Citation IDs (@item1) to include in the bibliography even if they are not cited in the document.

**Reply by timlrx** [reply]
In the `rehypePlugins` section of `contentlayer.config.ts` you can modify the line with `rehypeCitation` to something like: 
```ts
[rehypeCitation, { path: path.join(root, 'data'), noCite: ['@Xie2016'] }]
```
And in my blog post yaml, I add a `bibliography: references-data.bib` line. This adds the `Xie2016` bibliography reference to the blog post even if it is not cited. Note, this applies to **all** blog post that enable the bibliography option. If you want to control it at the post level, I could modify the plugin to do so similar to [pandoc](https://pandoc.org/chunkedhtml-demo/9.6-including-uncited-items-in-the-bibliography.html)

**Comment by timlrx** [comment, marked as answer]
@alexsalce try the updated version of rehype citation, v2.1.1. This supports customisation at the post level. You can add a `noCite` property to the frontmatter like the example below:

```md
---
title: 'Release of Tailwind Nextjs Starter Blog v2.0'
date: '2023-08-05'
lastmod: '2023-08-05'
tags: ['next-js', 'tailwind', 'guide', 'feature']
draft: false
summary: 'Release of Tailwind Nextjs Starter Blog template v2.0, refactored with Nextjs App directory and React Server Components setup.Discover the new features and how to migrate from V1.'
images: ['/static/images/twitter-card.png']
bibliography: references-data.bib
noCite: ["@Xie2016"]
---
```

Or use `noCite: ["@*"]` to display everything without citing.

**Reply by alexsalce** [reply]
This worked great, thank you! Very much appreciated!!
